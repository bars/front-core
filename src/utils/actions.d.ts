export interface PreparedActions {
  [key: string]: string
}

// TODO add redux typings
export interface Action {
  (dispatch?: Function, getState?: Function, option?: Object): void | Promise<any>
}

export const prepareActions: (actions: Array<string>, directory: string) => PreparedActions
export const onRequestSort: (fieldName: string) => Action
export const onChangeQuery: (field: string, value: any) => Action
export const onChangeFilter: (field: string, value: any) => Action
