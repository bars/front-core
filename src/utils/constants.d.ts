interface DataStates {
  [key: string]: string
}

export const tokenStorageKey: string
export const dataStates: DataStates
