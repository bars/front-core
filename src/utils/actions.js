import { deepAssign } from './lib-helpers'
import { getLocation } from '../selectors'

export const prepareActions = (actions, directory) => {
  const prefix = directory
    .replace(/^node_modules\//, '')
    .replace(/\//g, '-')

  return actions.reduce((acc, action) => ({
    ...acc,
    [action]: `${prefix}-${action}`,
  }), {})
}

export const onRequestSort = fieldName => (dispatch, getState) => {
  const { type, payload, query = {} } = getLocation(getState())
  const q = query.q || {}

  const orderBy = fieldName
  let orderDirection = 'desc'
  if (orderBy === q.orderBy && q.orderDirection === 'desc') {
    orderDirection = 'asc'
  }

  const nextQuery = deepAssign({}, query, { q: { orderBy, orderDirection } })
  dispatch({ type, payload, query: nextQuery })
}

export const onChangeQuery = (field, value) => (dispatch, getState) => {
  const { type, payload, query = {} } = getLocation(getState())

  const nextQuery = deepAssign({}, query, { q: { [field]: value } })
  dispatch({ type, payload, query: nextQuery })
}

export const onChangeFilter = (field, value) => (dispatch) => {
  dispatch(onChangeQuery('filter', { [field]: value }))
}

// deprecated
export default prepareActions
