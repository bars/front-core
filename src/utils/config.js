window.qa = window.qa || {}
const { qa } = window

qa.cacheExpire = 10 * 60 * 1000 // ten minutes

export default prop => qa[prop]
