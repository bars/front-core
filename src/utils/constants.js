export const tokenStorageKey = 'front-core-auth'
export const dataStates = {
  notAsked: 'notAsked',
  loading: 'loading',
  loaded: 'loaded',
  failed: 'failed',
  deprecated: 'deprecated',
  notFound: 'notFound',
}
