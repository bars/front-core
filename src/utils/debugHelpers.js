
// import { compare } from 'src/utils/debugHelpers'
//   shouldComponentUpdate(next) {
//     return compare(this.props, next, debug || console.log)
//   }

export const compare = (curr, next, debug) => {
  if (curr === next) return false
  const wrong = Object.keys(next)
    .filter((key) => {
      if (next[key] !== curr[key]) {
        debug('--- wrong ---', key, next[key], curr[key])
        return true
      }
      return false
    })
  return Boolean(wrong.length)
}
