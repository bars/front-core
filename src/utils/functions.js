import { createSelector, defaultMemoize } from 'reselect'

export const create = createSelector
export const memoize = defaultMemoize
