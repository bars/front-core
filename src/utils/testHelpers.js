import React from 'react'
import renderer from 'react-test-renderer'
import { JssProvider, SheetsRegistry } from 'react-jss'

global.requestAnimationFrame = (callback) => {
  setTimeout(callback, 0)
}
Date.now = jest.fn(() => 1000000000000)
Math.random = jest.fn(() => 0.1)
const genName = (rule, sheet) => {
  const m = (sheet && sheet.options.name) || 'm'
  return `${m}-${rule.key}`
}

const defaultFn = Component => (name, props) => {
  describe(`snapshot: ${name}`, () => {
    const sheets = new SheetsRegistry()
    const snap = renderer.create(
      <JssProvider generateClassName={genName} registry={sheets}>
        <Component {...props} />
      </JssProvider>
    )
    it('css', () => {
      expect(sheets.toString()).toMatchSnapshot()
    })
    it('html', () => {
      expect(snap.toJSON()).toMatchSnapshot()
    })
  })
}
const action = () => Function.prototype

const previewRunner = (dir, fn) => {
  jest.mock('react-dom')

  // https://github.com/facebook/react/issues/11565
  const ReactDOM = require('react-dom')
  ReactDOM.createPortal = node => node

  let previewDataLoaded
  try {
    previewDataLoaded = require(`${dir}/index.preview.js`).default
  } catch (_) {
    previewDataLoaded = [() => ({
      _preview: 'empty',
    })]
  }
  const previewData = [
    () => ({
      _preview: 'className',
      className: 'test-class',
    }),
    ...previewDataLoaded,
  ]

  let runner
  if (fn) {
    runner = fn
  } else {
    runner = defaultFn(require(dir).default)
  }

  previewData.forEach((getter) => {
    const { _preview, ...props } = getter({ action })
    runner(_preview, props)
  })
}

const root = `${__ROOT__.replace(/\//g, '-')}-`
const reducerRunner = ({ list, reducer, mock }) => {
  const all = ['fc-not-exist-type'].concat(list)
  let state
  all.forEach((type) => {
    const clientType = type.replace(root, '')
    test(`snapshot: ${clientType}`, () => {
      const payload = mock[clientType] || {}
      state = reducer(state, { ...payload, type })
      expect(state).toMatchSnapshot()
    })
  })
}

export {
  previewRunner,
  reducerRunner,
}
