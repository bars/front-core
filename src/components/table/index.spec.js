import { previewRunner } from 'src/utils/testHelpers'

previewRunner(__dirname)
