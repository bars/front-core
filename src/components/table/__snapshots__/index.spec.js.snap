// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot: className css 1`] = `
".MuiTypography-root {
  margin: 0;
  display: block;
}
.MuiTypography-display4 {
  color: rgba(0, 0, 0, 0.54);
  font-size: 7rem;
  font-weight: 300;
  font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;
  line-height: 1.14286em;
  margin-left: -.04em;
  letter-spacing: -.04em;
}
.MuiTypography-display3 {
  color: rgba(0, 0, 0, 0.54);
  font-size: 3.5rem;
  font-weight: 400;
  font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;
  line-height: 1.30357em;
  margin-left: -.02em;
  letter-spacing: -.02em;
}
.MuiTypography-display2 {
  color: rgba(0, 0, 0, 0.54);
  font-size: 2.8125rem;
  font-weight: 400;
  font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;
  line-height: 1.06667em;
  margin-left: -.02em;
}
.MuiTypography-display1 {
  color: rgba(0, 0, 0, 0.54);
  font-size: 2.125rem;
  font-weight: 400;
  font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;
  line-height: 1.20588em;
}
.MuiTypography-headline {
  color: rgba(0, 0, 0, 0.87);
  font-size: 1.5rem;
  font-weight: 400;
  font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;
  line-height: 1.35417em;
}
.MuiTypography-title {
  color: rgba(0, 0, 0, 0.87);
  font-size: 1.3125rem;
  font-weight: 500;
  font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;
  line-height: 1.16667em;
}
.MuiTypography-subheading {
  color: rgba(0, 0, 0, 0.87);
  font-size: 1rem;
  font-weight: 400;
  font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;
  line-height: 1.5em;
}
.MuiTypography-body2 {
  color: rgba(0, 0, 0, 0.87);
  font-size: 0.875rem;
  font-weight: 500;
  font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;
  line-height: 1.71429em;
}
.MuiTypography-body1 {
  color: rgba(0, 0, 0, 0.87);
  font-size: 0.875rem;
  font-weight: 400;
  font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;
  line-height: 1.46429em;
}
.MuiTypography-caption {
  color: rgba(0, 0, 0, 0.54);
  font-size: 0.75rem;
  font-weight: 400;
  font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;
  line-height: 1.375em;
}
.MuiTypography-button {
  font-size: 0.875rem;
  font-weight: 500;
  font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;
  text-transform: uppercase;
}
.MuiTypography-alignLeft {
  text-align: left;
}
.MuiTypography-alignCenter {
  text-align: center;
}
.MuiTypography-alignRight {
  text-align: right;
}
.MuiTypography-alignJustify {
  text-align: justify;
}
.MuiTypography-noWrap {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}
.MuiTypography-gutterBottom {
  margin-bottom: 0.35em;
}
.MuiTypography-paragraph {
  margin-bottom: 16px;
}
.MuiTypography-colorInherit {
  color: inherit;
}
.MuiTypography-colorPrimary {
  color: #3f51b5;
}
.MuiTypography-colorSecondary {
  color: #f50057;
}
.MuiTypography-colorTextSecondary {
  color: rgba(0, 0, 0, 0.54);
}
.MuiTypography-colorError {
  color: #f44336;
}

.fcTable-root a {
  color: #ff4081;
  text-decoration: none;
}
.fcTable-root a:visited {
  color: #ff4081;
}
.fcTable-nothing {
  margin-top: 48px;
}
.fcTable-cell {
  white-space: normal;
  padding-top: 8px;
  padding-bottom: 8px;
}
.fcTable-highlight {
  background-color: #e8eaf6;
}
.fcTable-pointer {
  cursor: pointer;
}"
`;

exports[`snapshot: className html 1`] = `
<h3
  className="MuiTypography-root MuiTypography-subheading MuiTypography-alignCenter test-class fcTable-nothing fcTypography-root"
>
  No entries to display
</h3>
`;

exports[`snapshot: with text and type css 1`] = `
".MuiTable-root {
  width: 100%;
  display: table;
  font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;
  border-spacing: 0;
  border-collapse: collapse;
}
.MuiTableBody-root {
  display: table-row-group;
}
.MuiTableCell-root {
  display: table-cell;
  padding: 4px 56px 4px 24px;
  text-align: left;
  border-bottom: 1px solid
    rgba(224, 224, 224, 1);
  vertical-align: inherit;
}
.MuiTableCell-root:last-child {
  padding-right: 24px;
}
.MuiTableCell-head {
  color: rgba(0, 0, 0, 0.54);
  font-size: 0.75rem;
  font-weight: 500;
}
.MuiTableCell-body {
  color: rgba(0, 0, 0, 0.87);
  font-size: 0.8125rem;
}
.MuiTableCell-footer {
  color: rgba(0, 0, 0, 0.54);
  font-size: 0.75rem;
  border-bottom: 0;
}
.MuiTableCell-numeric {
  text-align: right;
  flex-direction: row-reverse;
}
.MuiTableCell-paddingDense {
  padding-right: 24px;
}
.MuiTableCell-paddingCheckbox {
  padding: 0 12px;
}
.MuiTableCell-paddingNone {
  padding: 0;
}
.MuiTableCell-paddingNone:last-child {
  padding: 0;
}
.MuiTableHead-root {
  display: table-header-group;
}
.MuiPaper-root {
  background-color: #fff;
}
.MuiPaper-rounded {
  border-radius: 2px;
}
.MuiPaper-elevation0 {
  box-shadow: none;
}
.MuiPaper-elevation1 {
  box-shadow: 0px 1px 3px 0px rgba(0, 0, 0, 0.2),0px 1px 1px 0px rgba(0, 0, 0, 0.14),0px 2px 1px -1px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation2 {
  box-shadow: 0px 1px 5px 0px rgba(0, 0, 0, 0.2),0px 2px 2px 0px rgba(0, 0, 0, 0.14),0px 3px 1px -2px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation3 {
  box-shadow: 0px 1px 8px 0px rgba(0, 0, 0, 0.2),0px 3px 4px 0px rgba(0, 0, 0, 0.14),0px 3px 3px -2px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation4 {
  box-shadow: 0px 2px 4px -1px rgba(0, 0, 0, 0.2),0px 4px 5px 0px rgba(0, 0, 0, 0.14),0px 1px 10px 0px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation5 {
  box-shadow: 0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 5px 8px 0px rgba(0, 0, 0, 0.14),0px 1px 14px 0px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation6 {
  box-shadow: 0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 6px 10px 0px rgba(0, 0, 0, 0.14),0px 1px 18px 0px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation7 {
  box-shadow: 0px 4px 5px -2px rgba(0, 0, 0, 0.2),0px 7px 10px 1px rgba(0, 0, 0, 0.14),0px 2px 16px 1px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation8 {
  box-shadow: 0px 5px 5px -3px rgba(0, 0, 0, 0.2),0px 8px 10px 1px rgba(0, 0, 0, 0.14),0px 3px 14px 2px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation9 {
  box-shadow: 0px 5px 6px -3px rgba(0, 0, 0, 0.2),0px 9px 12px 1px rgba(0, 0, 0, 0.14),0px 3px 16px 2px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation10 {
  box-shadow: 0px 6px 6px -3px rgba(0, 0, 0, 0.2),0px 10px 14px 1px rgba(0, 0, 0, 0.14),0px 4px 18px 3px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation11 {
  box-shadow: 0px 6px 7px -4px rgba(0, 0, 0, 0.2),0px 11px 15px 1px rgba(0, 0, 0, 0.14),0px 4px 20px 3px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation12 {
  box-shadow: 0px 7px 8px -4px rgba(0, 0, 0, 0.2),0px 12px 17px 2px rgba(0, 0, 0, 0.14),0px 5px 22px 4px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation13 {
  box-shadow: 0px 7px 8px -4px rgba(0, 0, 0, 0.2),0px 13px 19px 2px rgba(0, 0, 0, 0.14),0px 5px 24px 4px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation14 {
  box-shadow: 0px 7px 9px -4px rgba(0, 0, 0, 0.2),0px 14px 21px 2px rgba(0, 0, 0, 0.14),0px 5px 26px 4px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation15 {
  box-shadow: 0px 8px 9px -5px rgba(0, 0, 0, 0.2),0px 15px 22px 2px rgba(0, 0, 0, 0.14),0px 6px 28px 5px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation16 {
  box-shadow: 0px 8px 10px -5px rgba(0, 0, 0, 0.2),0px 16px 24px 2px rgba(0, 0, 0, 0.14),0px 6px 30px 5px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation17 {
  box-shadow: 0px 8px 11px -5px rgba(0, 0, 0, 0.2),0px 17px 26px 2px rgba(0, 0, 0, 0.14),0px 6px 32px 5px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation18 {
  box-shadow: 0px 9px 11px -5px rgba(0, 0, 0, 0.2),0px 18px 28px 2px rgba(0, 0, 0, 0.14),0px 7px 34px 6px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation19 {
  box-shadow: 0px 9px 12px -6px rgba(0, 0, 0, 0.2),0px 19px 29px 2px rgba(0, 0, 0, 0.14),0px 7px 36px 6px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation20 {
  box-shadow: 0px 10px 13px -6px rgba(0, 0, 0, 0.2),0px 20px 31px 3px rgba(0, 0, 0, 0.14),0px 8px 38px 7px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation21 {
  box-shadow: 0px 10px 13px -6px rgba(0, 0, 0, 0.2),0px 21px 33px 3px rgba(0, 0, 0, 0.14),0px 8px 40px 7px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation22 {
  box-shadow: 0px 10px 14px -6px rgba(0, 0, 0, 0.2),0px 22px 35px 3px rgba(0, 0, 0, 0.14),0px 8px 42px 7px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation23 {
  box-shadow: 0px 11px 14px -7px rgba(0, 0, 0, 0.2),0px 23px 36px 3px rgba(0, 0, 0, 0.14),0px 9px 44px 8px rgba(0, 0, 0, 0.12);
}
.MuiPaper-elevation24 {
  box-shadow: 0px 11px 15px -7px rgba(0, 0, 0, 0.2),0px 24px 38px 3px rgba(0, 0, 0, 0.14),0px 9px 46px 8px rgba(0, 0, 0, 0.12);
}
.MuiButtonBase-root {
  color: inherit;
  cursor: pointer;
  margin: 0;
  border: 0;
  display: inline-flex;
  padding: 0;
  outline: none;
  position: relative;
  user-select: none;
  align-items: center;
  border-radius: 0;
  vertical-align: middle;
  justify-content: center;
  -moz-appearance: none;
  text-decoration: none;
  background-color: transparent;
  -webkit-appearance: none;
  -webkit-tap-highlight-color: transparent;
}
.MuiButtonBase-root::-moz-focus-inner {
  border-style: none;
}
.MuiButtonBase-root.MuiButtonBase-disabled {
  cursor: default;
  pointer-events: none;
}
.MuiSvgIcon-root {
  fill: currentColor;
  width: 1em;
  height: 1em;
  display: inline-block;
  font-size: 24px;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  user-select: none;
  flex-shrink: 0;
}
.MuiSvgIcon-colorPrimary {
  color: #3f51b5;
}
.MuiSvgIcon-colorSecondary {
  color: #f50057;
}
.MuiSvgIcon-colorAction {
  color: rgba(0, 0, 0, 0.54);
}
.MuiSvgIcon-colorDisabled {
  color: rgba(0, 0, 0, 0.26);
}
.MuiSvgIcon-colorError {
  color: #f44336;
}
.MuiTableRow-root {
  color: inherit;
  height: 48px;
  display: table-row;
  vertical-align: middle;
}
.MuiTableRow-root:focus {
  outline: none;
}
.MuiTableRow-root.MuiTableRow-selected {
  background-color: rgba(0, 0, 0, 0.04);
}
.MuiTableRow-root.MuiTableRow-hover:hover {
  background-color: rgba(0, 0, 0, 0.07);
}
.MuiTableRow-head {
  height: 56px;
}
.MuiTableRow-footer {
  height: 56px;
}
.MuiTableSortLabel-root {
  cursor: pointer;
  display: inline-flex;
  align-items: center;
  flex-direction: inherit;
  justify-content: flex-start;
}
.MuiTableSortLabel-root:hover {
  color: rgba(0, 0, 0, 0.87);
}
.MuiTableSortLabel-root:focus {
  color: rgba(0, 0, 0, 0.87);
}
.MuiTableSortLabel-active {
  color: rgba(0, 0, 0, 0.87);
}
.MuiTableSortLabel-active .MuiTableSortLabel-icon {
  opacity: 1;
}
.MuiTableSortLabel-icon {
  width: 16px;
  height: 16px;
  opacity: 0;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  margin-left: 4px;
  user-select: none;
  margin-right: 4px;
}
.MuiTableSortLabel-iconDirectionDesc {
  transform: rotate(0deg);
}
.MuiTableSortLabel-iconDirectionAsc {
  transform: rotate(180deg);
}

.fcTableHead-center {
  text-align: center;
}"
`;

exports[`snapshot: with text and type css 2`] = `""`;

exports[`snapshot: with text and type html 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation4 MuiPaper-rounded fcTable-root fcPaper-root"
>
  <table
    className="MuiTable-root"
  >
    <thead
      className="MuiTableHead-root fcTableHead-root"
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          <span
            className="MuiButtonBase-root MuiTableSortLabel-root"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            Key
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </span>
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          <span
            className="MuiButtonBase-root MuiTableSortLabel-root"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            Title
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </span>
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
    >
      <tr
        className="MuiTableRow-root MuiTableRow-hover"
        onClick={[Function]}
        tabIndex={-1}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root fcTable-cell MuiTableCell-body"
          scope={undefined}
        >
          key-0
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root fcTable-cell MuiTableCell-body"
          scope={undefined}
        >
          one
        </td>
      </tr>
      <tr
        className="MuiTableRow-root MuiTableRow-hover"
        onClick={[Function]}
        tabIndex={-1}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root fcTable-cell MuiTableCell-body"
          scope={undefined}
        >
          key-1
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root fcTable-cell MuiTableCell-body"
          scope={undefined}
        >
          two
        </td>
      </tr>
      <tr
        className="MuiTableRow-root MuiTableRow-hover fcTable-highlight"
        onClick={[Function]}
        tabIndex={-1}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root fcTable-cell MuiTableCell-body"
          scope={undefined}
        >
          key-2
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root fcTable-cell MuiTableCell-body"
          scope={undefined}
        >
          three
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`snapshot: with text and type html 2`] = `
<div
  className="MuiPaper-root MuiPaper-elevation4 MuiPaper-rounded fcTable-root fcPaper-root"
>
  <table
    className="MuiTable-root"
  >
    <thead
      className="MuiTableHead-root fcTableHead-root"
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          <span
            className="MuiButtonBase-root MuiTableSortLabel-root"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            Key
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </span>
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          <span
            className="MuiButtonBase-root MuiTableSortLabel-root MuiTableSortLabel-active"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            Title
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </span>
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
    >
      <tr
        className="MuiTableRow-root MuiTableRow-hover"
        onClick={[Function]}
        tabIndex={-1}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root fcTable-cell MuiTableCell-body"
          scope={undefined}
        >
          key-0
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root fcTable-cell MuiTableCell-body"
          scope={undefined}
        >
          one
        </td>
      </tr>
      <tr
        className="MuiTableRow-root MuiTableRow-hover"
        onClick={[Function]}
        tabIndex={-1}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root fcTable-cell MuiTableCell-body"
          scope={undefined}
        >
          key-1
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root fcTable-cell MuiTableCell-body"
          scope={undefined}
        >
          two
        </td>
      </tr>
      <tr
        className="MuiTableRow-root MuiTableRow-hover fcTable-highlight"
        onClick={[Function]}
        tabIndex={-1}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root fcTable-cell MuiTableCell-body"
          scope={undefined}
        >
          key-2
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root fcTable-cell MuiTableCell-body"
          scope={undefined}
        >
          three
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`snapshot: without props css 1`] = `""`;

exports[`snapshot: without props html 1`] = `
<h3
  className="MuiTypography-root MuiTypography-subheading MuiTypography-alignCenter fcTable-nothing fcTypography-root"
>
  No entries to display
</h3>
`;
